public/js/chatService.js

```diff
@@ -0,0 +1,95 @@
+(function() {
+    var app = angular.module("ChatApp");
+    app.service("chatService", ["$http", function ($http) {
+
+        var self = this;
```

Comment:
since you are using this assignment, it is worth being consistent, and using self at the top level instead of this.